File: src/main/java/um/haberes/rest/model/view/ImputadoSueldo.java
/**
 * 
 */
package um.haberes.rest.model.view;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;

import org.hibernate.annotations.Immutable;

import um.haberes.rest.model.view.pk.ImputadoSueldoPk;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Immutable
@Table(name = "vw_imputado_sueldo")
@IdClass(ImputadoSueldoPk.class)
@NoArgsConstructor
@AllArgsConstructor
public class ImputadoSueldo implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -5028212004561357534L;
	
	@Id
	private Integer anho;
	
	@Id
	private Integer mes;
	
	@Id
	private BigDecimal cuentaSueldos;
	
	private BigDecimal totalImputado;

}

File: src/main/java/um/haberes/rest/kotlin/model/view/ImputadoSueldo.kt
package um.haberes.rest.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import org.hibernate.annotations.Immutable
import java.math.BigDecimal

@Entity
@Table(name = "vw_imputado_sueldo",
    uniqueConstraints = [UniqueConstraint(columnNames = ["anho", "mes", "cuentaSueldos"])])
@Immutable
data class ImputadoSueldo(

    @Id
    var uniqueId: String? = null,

    var anho: Int? = null,
    var mes: Int? = null,
    var cuentaSueldos: BigDecimal? = null,
    var totalImputado: BigDecimal? = null

)
